modules/search/class-jetpack-instant-search.php

```diff
@@ -0,0 +1,367 @@
+	 *
+	 * @return object|WP_Error The response from the public API, or a WP_Error.
+	 */
+	public function instant_api( array $args ) {
+		$start_time = microtime( true );
+
+		// Cache locally to avoid remote request slowing the page.
+		$transient_name = '_jetpack_instant_search_cache_' . md5( wp_json_encode( $args ) );
+		$cache          = get_transient( $transient_name );
+		if ( false !== $cache ) {
+			$end_time = microtime( true );
+			return $cache;
+		}
+
+		$endpoint     = sprintf( '/sites/%s/search', $this->jetpack_blog_id );
+		$query_params = urldecode( http_build_query( $args ) );
+		$service_url  = 'https://public-api.wordpress.com/rest/v1.3' . $endpoint . '?' . $query_params;
+
+		$request_args = array(
+			'timeout'    => 10,
+			'user-agent' => 'jetpack_search',
+		);
+
+		$request  = wp_remote_get( $service_url, $request_args );
+		$end_time = microtime( true );
+
+		if ( is_wp_error( $request ) ) {
+			return $request;
+		}
+
+		$response_code = wp_remote_retrieve_response_code( $request );
+		$response      = json_decode( wp_remote_retrieve_body( $request ), true );
+
+		if ( ! $response_code || $response_code < 200 || $response_code >= 300 ) {
+			/**
+			 * Fires after a search query request has failed
+			 *
+			 * @module search
+			 *
+			 * @since  5.6.0
+			 *
+			 * @param array Array containing the response code and response from the failed search query
+			 */
+			do_action(
+				'failed_jetpack_search_query',
+				array(
+					'response_code' => $response_code,
+					'json'          => $response,
+				)
+			);
+
+			return new WP_Error( 'invalid_search_api_response', 'Invalid response from API - ' . $response_code );
+		}
+
+		$took = is_array( $response ) && ! empty( $response['took'] )
+			? $response['took']
+			: null;
+
+		$query = array(
+			'args'          => $args,
+			'response'      => $response,
+			'response_code' => $response_code,
+			'elapsed_time'  => ( $end_time - $start_time ) * 1000, // Convert from float seconds to ms.
+			'es_time'       => $took,
+			'url'           => $service_url,
+		);
+
+		/**
+		 * Fires after a search request has been performed.
+		 *
+		 * Includes the following info in the $query parameter:
+		 *
+		 * array args Array of Elasticsearch arguments for the search
+		 * array response Raw API response, JSON decoded
+		 * int response_code HTTP response code of the request
+		 * float elapsed_time Roundtrip time of the search request, in milliseconds
+		 * float es_time Amount of time Elasticsearch spent running the request, in milliseconds
+		 * string url API url that was queried
+		 *
+		 * @module search
+		 *
+		 * @since  5.0.0
+		 * @since  5.8.0 This action now fires on all queries instead of just successful queries.
+		 *
+		 * @param array $query Array of information about the query performed
+		 */
+		do_action( 'did_jetpack_search_query', $query );
+
+		// Update local cache.
+		set_transient( $transient_name, $response, 1 * HOUR_IN_SECONDS );
```

Comment:
Why are we having a 1 hr cache on the response? I would assume that with sites with large amounts of content modifications (news) that this would have stale content in results often. 

Shouldn't the cache expiration match the WP.com endpoint? or be 5/10 minutes max?

Reply:
cc @gibrown

Reply:
This query is only used for displaying filters/facets, and it runs on every page load. So my thinking here is that the filter counts may end up sometimes being off a bit, but by having a long-ish cache time we should be able to avoid adding extra latency to every page load.